### PR TITLE
First-steps

### PR DESCRIPTION
**Pull request by xtarzyk**

### CODE REVIEW

File: ingredients.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+const ingredientsSet = new Set()
+// ingredients = localStorage.getItem('ingredients')
```
Comment (jpudysz) [inline ingredients.js]: remove commented code

File: ingredients.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+// ingredients.add(setIng)
+const createSpanBtns = () => {
+    const $pencilIcon = $('<i>').addClass('fa-solid fa-pencil')
+    const $trashIcon = $('<i>').addClass('fa-solid fa-trash-can')
+    
+    $trashIcon.click(event => {
+        removeIngredient(event)
+        $(event.target).parentsUntil($('.content__list')).remove()
+      })
```
Comment (jpudysz) [inline ingredients.js]: bad indentation

File: ingredients.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    $pencilIcon.click(event => {
```
Comment (jpudysz) [inline ingredients.js]: proxy, this is equivalent to:
```
$pencilIcon.click(editIngredient)
```

File: ingredients.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        editIngredient(event)
+    })
+    
+    return $('<span>').append($pencilIcon, $trashIcon)
+}
+
+export const getLocalStorageData = () => {
+    if (localStorage.getItem('ingredients') !== null) {
+        const receivedIngredients = localStorage.getItem('ingredients').split(',')
+        console.log(receivedIngredients)
```
Comment (jpudysz) [inline ingredients.js]: enter between consts and lets (assignments) and instructions

File: ingredients.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    $pencilIcon.click(event => {
+        editIngredient(event)
+    })
+    
+    return $('<span>').append($pencilIcon, $trashIcon)
+}
+
+export const getLocalStorageData = () => {
+    if (localStorage.getItem('ingredients') !== null) {
+        const receivedIngredients = localStorage.getItem('ingredients').split(',')
```
Comment (jpudysz) [inline ingredients.js]: you can move localStorage logic to the next file and abstract it away. If you do it you don't need to parse everything repeatedly

File: ingredients.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        receivedIngredients.forEach(ing => {
+            ingredientsSet.add(ing)
+            createIngredients()
+        })
+    }
+}
+
+export const updateIngredientsList = () => {
+    ingredientsSet.delete('')
+    ingredientsSet.delete(undefined)
```
Comment (jpudysz) [inline ingredients.js:29]: ?

File: ingredients.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    $('<div>').addClass('content__list').appendTo('.content')
+}
+
+export const createIngredients = () => {
+    ingredientsSet.add($('.content__input').val())
+    updateIngredientsList()
+
+    ingredientsSet.forEach(text => {
+        const $newListItem = $('<div>').addClass('content__list-item')
+        const $spanBtns = createSpanBtns()
```
Comment (jpudysz) [inline ingredients.js:43]: aa

File: ingredients.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        console.log(text)
+        $newListItem
+            .text(text)
+            .append($spanBtns)
+            .appendTo($('.content__list'))
+    })
+}
+
+const removeIngredient = event => {
+    const selectedItem = $(event.target).closest('div').text()
```
Comment (jpudysz) [inline ingredients.js:53]: aa

File: main.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+const $input = $('<input type="text">').addClass('content__input')
+const $addingBtn = $('<span>').text('+').addClass('adding-btn')
+
+const createIngredientsInput = () => {
+  $('.content__header').append($input.attr('placeholder', 'Add ingredient...'))
+  $input.change(() => {
+    createIngredients()
+    $input.val('')
+  })
+
```
Comment (jpudysz) [inline main.js]: remove enter

File: recipes.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+let selectedIngredients = []
+
+const createTrashBtn = () => {
+    const $trashIcon = $('<i>').addClass('fa-solid fa-trash-can ing-can')
+    
+    // $trashIcon.click(event => {
```
Comment (jpudysz) [inline recipes.js]: remove commented code

File: recipes.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+                .text(ing)
+                .click(selectIng)
+                .appendTo($ingredientList)
+        })
+    }
+}
+
+const selectIng = event => {
+    $(event.target).css('background', '#243535')
+    if (selectedIngredients.includes($(event.target).text())) {
```
Comment (jpudysz) [inline recipes.js:25]: enter

File: recipes.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    selectedIngredients = selectedIngredients.concat($(event.target).text())
+    console.log(selectedIngredients)
+}
+
+const createRecipe = () => {
+    if(selectedIngredients.length === 0) {
+        return
+    }
+    const $recipesList = $('<div>').addClass('content__recipes-list').appendTo('.content')
```
Comment (jpudysz) [inline recipes.js:41]: enter

File: recipes.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (jpudysz) [inline recipes.js]: remove enter

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: ingredients.js (hunk 6 above)
Comment (xtarzyk) [reply ingredients.js:29]: I don't know why and how, but at the some moment these two above are creating in set

File: ingredients.js [hunk 14]
```diff
@@ -8,28 +10,25 @@ const createSpanBtns = () => {
...
-    if (localStorage.getItem('ingredients') !== null) {
-        const receivedIngredients = localStorage.getItem('ingredients').split(',')
-        
-        console.log(receivedIngredients)
-        receivedIngredients.forEach(ing => {
-            ingredientsSet.add(ing)
-            createIngredients()
-        })
-    }
+export const createIngsListFromLocalStorage = () => {
```
Comment (jpudysz) [inline ingredients.js]: using shortcut for names is a bad practice

File: ingredients.js [hunk 15]
```diff
@@ -8,28 +10,25 @@ const createSpanBtns = () => {
...
+    const receivedIngredients = getLocalStorageData()
+
+    console.log(receivedIngredients)
+    receivedIngredients.forEach(ing => {
```
Comment (jpudysz) [inline ingredients.js]: same here, `ingredients` will be fine

File: recipes.js [hunk 16]
```diff
@@ -1,38 +1,34 @@
...
-                .text(ing)
-                .click(selectIng)
-                .appendTo($ingredientList)
-        })
-    }
+export const showIngredients = () => {
+    const receivedIngredients = getLocalStorageData()
+    const $ingredientList = $('<div>').addClass('content__ingredients-list').appendTo('.content')
+    
+    receivedIngredients.forEach(ing => {
```
Comment (jpudysz) [inline recipes.js]: ingredients

File: utils.js [hunk 17]
```diff
@@ -0,0 +1,11 @@
+export const getLocalStorageData = () => {
```
Comment (jpudysz) [inline utils.js]: you can update this file to new version

**Review by jpudysz — CHANGES_REQUESTED** [review]

**Review by jpudysz — APPROVED** [review]

**Pull request merged**